auto sync 2022-03-09

nunnatsa opened the pull request:
auto sync upstream 2022-03-09

nirarg commented:
/bugzilla refresh

nirarg commented:
/bugzilla refresh

bardielle commented on Dockerfile.rhel:
```diff
@@ -13,15 +13,20 @@
 WORKDIR /go/src/github.com/kubevirt/cluster-api-provider-kubevirt
 COPY . .
-RUN make build NO_DOCKER=1
+RUN NO_DOCKER=1 make build
```
why does it matter?

nirarg replied:
Removed Dockerfile changes

bardielle commented on cmd/manager/main.go:
```diff
@@ -136,7 +128,7 @@ func main() {
 
 	// Initialize infra-cluster clients
-	infraClusterClient, err := infracluster.New(tenantClusterClient)
+	infraClusterClient, err := infracluster.New(context.Background(), tenantClusterClient)
```
Dont think it's related to that commit

nirarg replied:
Changed the commits

bardielle commented on cmd/manager/main.go:
```diff
@@ -42,23 +43,19 @@ var (
-
-	// TODO Add relevant flags as written in klog initFlags
-	//klog.InitFlags(nil)
+	klog.InitFlags(nil)
```
I think that this is the rout cause of your error - can you test it without the last commit but only with that?

nirarg replied:
Removed Dockerfile changes and validated

nirarg commented on pkg/clients/infracluster/client.go:
```diff
@@ -41,21 +44,30 @@ const (
 }
 
+var (
+	vmRes = schema.GroupVersionResource{
```
Changed

bardielle replied:
Can you squash you last commit with that commit?

nirarg commented on pkg/clients/infracluster/client.go:
```diff
@@ -41,21 +44,30 @@ const (
+		Version:  kubevirtapiv1.GroupVersion.Version,
+		Resource: "virtualmachines",
+	}
+	vmiRes = schema.GroupVersionResource{
```
Changed

nirarg commented on pkg/clients/infracluster/client.go:
```diff
@@ -89,52 +97,128 @@ func New(tenantClusterKubernetesClient tenantcluster.Client) (Client, error) {
+		return nil, errors.Wrap(err, "failed to list VirtualMachine")
+	}
+	var vmList kubevirtapiv1.VirtualMachineList
+	err = c.fromUnstructedListToInterface(*resp, &vmList, "VirtualMachineList")
```
vmList is not a pointer, so its default value is empty struct (can't be nil)

nirarg commented:
/bugzilla refresh

nirarg commented:
/bugzilla refresh

nirarg commented:
@JoelSpeed this PR bump `machine-api-operator` version and removing the replacement

bardielle reviewed (approved):
/lgtm

nirarg committed:
Change OWNERS file

nunnatsa committed:
Automatic sync with upstream

This PR adds a new github action to daily sync with upstream. It checks if there are new changes in upstream, and if there are, it creates a PR to sync.

The new PR is created by hard reseting to the upstream main branch, then restoring the downstream modified files.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
add script to sync with upstream

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
create PR

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
Move to ginkgo v2

Moving to ginkgo v2 allows filtering by labels.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

k8s-ci-robot committed:
Merge pull request #109 from nunnatsa/move-to-ginkgov2

Move to ginkgo v2

davidvossel reviewed (approved):
/lgtm
/approve
